x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -105,7 +113,13 @@ export function getTransactionType({
     : TRANSACTION_REQUEST;
 
   // If the default transaction type is not in transactionTypes the first in the list is returned
-  return transactionTypes.includes(defaultTransactionType)
+  const currentTransactionType = transactionTypes.includes(
+    defaultTransactionType
+  )
     ? defaultTransactionType
     : transactionTypes[0];
+
+  // Replace transactionType in the URL in case it is not one of the types returned by the API
+  replace(history, { query: { transactionType: currentTransactionType } });
```

Comment:
I don't think a function called `getTransactionType` should trigger a redirect. Can we move this into a component or rename this function to something like `getOrRedirectToTransactionType` that makes it clear that this is a function with side effects?

Reply:
OK, that sounds good to me. But it looks like there is just one component using this right?

Reply:
The `ApmServiceContextProvider` uses it, which is used by the apm template